### DISCUSSION TITLE
Unable to configure nextjs to run with nginx server in docker

### DISCUSSION THREAD

**Discussion by dMaisnam** [vercel/next.js#31017] [Help] [answered]
I am trying to setup nginx server in my dockerized nextjs app. The main `Dockerfile` & `docker-compose.yml` files live in project root while the nginx files live in `nginx/default.conf` and `nginx/Dockerfile`. 

When I run docker compose, the nginx server exits with the message `nginx: [emerg] host not found in upstream "nextjs:3000" in /etc/nginx/conf.d/default.conf:4`. 

Here are how my files look (code uploaded as an image to reduce length. Can upload as text if required):

![nextjs-docker](https://user-images.githubusercontent.com/32391506/140492803-1881d75a-720b-4040-b652-3891de930cdb.png)
- LEFT: `Dockerfile`
- MIDDLE: `nginx/default.conf`
- RIGHT-TOP: `docker-compose.yml`
- RIGHT-BOTTOM: `nginx/Dockerfile`

The nextjs server runs fine on `localhost:3000`. Here are some resources I referred to
- https://hkc7180.medium.com/nextjs-with-nginx-on-ec2-by-docker-compose-7558c838e0a4
- https://steveholgado.com/nginx-for-nextjs/
- https://github.com/vercel/next.js/tree/canary/examples/with-docker

This is quite new to me and I am unable to figure out how to forward nginx server for the nextjs frontend. How do I fix this?
Any help is much appreciated. Thank you for your time.

**Comment by wwwdepot** [comment]
You don't have `nextjs` service name defined in docker-compose file. You have `app` and `nginx`.

**Reply by dMaisnam** [reply]
> You don't have `nextjs` service name defined in docker-compose file. You have `app` and `nginx`.

Thanks for the suggestion. So I need to change service name to `nextjs` from `app`? I will give this a try.

**Comment by dMaisnam** [comment, marked as answer]
From @wwwdepot 's answer, I figured I needed to use the nextjs container name `ifp-frontend` as the server name of nginx. Changing the `server nextjs:3000;` to `server ifp-frontend:3000;` in `nginx/default.conf` fixed my problem.
